File: server/src/Routes/WorkRoutes.js
import express from "express";
import JWT from "../Middlewares/jsonwebtoken.js";
import WorkController from "../Controllers/workController.js";

const WorkRoutes = express.Router();

WorkRoutes.get(
  "/get-one/:workId",
  JWT.validateAccessToken,
  WorkController.getWork
);
WorkRoutes.get(
  "/get-user-works",
  JWT.validateAccessToken,
  WorkController.getUserWorks
);
WorkRoutes.post(
  "/add-work",
  JWT.validateAccessToken,
  WorkController.createWork
);
WorkRoutes.get(
  "/get-all",
  JWT.validateAccessToken,
  WorkController.getAllWorks
);
WorkRoutes.put(
  "/update/:workId",
  JWT.validateAccessToken,
  WorkController.updateWork
);
WorkRoutes.delete(
  "/delete/:workId",
  JWT.validateAccessToken,
  WorkController.deleteWork
);

export default WorkRoutes;

File: server/src/routes/workRoutes.go
package routes

import (
	"controll/src/controllers"

	"github.com/gin-gonic/gin"
)

func WorkRoutes(router *gin.RouterGroup) {
	work := router.Group("/api/work")
	{
		work.GET("/get-one/:workId", controllers.GetWork())
		work.GET("/get-user-works", controllers.GetUserWorks())
		work.POST("/create", controllers.CreateWork())
		work.GET("/get-all", controllers.GetAllWorks())
		work.PUT("/update/:workId", controllers.UpdateWork())
		work.DELETE("/delete/:workId", controllers.DeleteWork())
	}
}
